src/interpreter/interpreter.h

```diff
@@ -18,18 +18,24 @@
 #define interpreter_interpreter_h
 
 #include "store.h"
+#include "support/result.h"
 
 namespace wasm {
 
 class Interpreter {
 public:
   // TODO: Methods to instantiate modules.
   // TODO: Methods to run exported functions.
-  std::vector<Literal> run(Expression* root);
+
+  Result<> addInstance(std::shared_ptr<Module> wasm);
+  std::vector<Literal> runTest(Expression* root);
+  std::vector<Literal> run();
```

Comment:
Let's make these return `Literals`, which is a subclass of `SmallVector<Literal, 1>`, because by far the common case is that there will be just one value to return.

But it's fine if this is in a follow-up.

Reply:
Agreed, as a follow-up